### ISSUE TITLE
Make BlueCode more JSR-82 compliant

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Dorkasdev/bluecove#8]
```
As Vlad Skarzhevskyy wrote in a email to bluecove-developers:
> I run automatic comparison of BlueCove with one from sun WTK. The list 
> of differences follows. My suggestion is to implement all missing 
> function and if it is not implemented we should throw appropriate 
> Exception or Error. This would allow compilation and testing of MIDlet 
> applications with BlueCove library on windows. 
>
> javax.microedition.io.Connector.openInputStream is Missing 
> javax.microedition.io.Connector.openDataInputStream is Missing 
> javax.microedition.io.Connector.openOutputStream is Missing 
> javax.microedition.io.Connector.openDataOutputStream is Missing 
> javax.microedition.io.Connector.open is Missing 
> javax.bluetooth.DataElement.getValue getModifiers expected:<... 
> synchronized> but was:<...> 
> javax.bluetooth.DataElement.addElement getModifiers expected:<... 
> synchronized> but was:<...> 
> javax.bluetooth.DataElement.getSize getModifiers expected:<... 
> synchronized> but was:<...> 
> javax.bluetooth.DataElement.insertElementAt getModifiers expected:<... 
> synchronized> but was:<...> 
> javax.bluetooth.DiscoveryAgent.cancelServiceSearch is Missing 
> javax.bluetooth.DiscoveryAgent.selectService is Missing 
> javax.bluetooth.LocalDevice.updateRecord is Missing 
> javax.bluetooth.RemoteDevice.authorize is Missing 
> javax.bluetooth.RemoteDevice.encrypt is Missing 
> javax.bluetooth.RemoteDevice.authenticate is Missing 
> javax.bluetooth.BluetoothConnectionException not implemented 
> javax.bluetooth.ServiceRegistrationException not implemented 
> 
> Not required. 
> javax.bluetooth.L2CAPConnection not implemented 
> javax.bluetooth.L2CAPConnectionNotifier not implemented
```

Original issue reported on code.google.com by `paul.tot...@gmail.com` on 4 Dec 2006 at 3:41

**Comment by GoogleCodeExporter** [comment]
```
I've reviewed the changes and am now waiting for confirmation by Vlad that I 
haven't 
broken anything.
```

Original comment by `paul.tot...@gmail.com` on 15 Dec 2006 at 11:21

**Comment by GoogleCodeExporter** [comment]
```
I'm still very new to this bugtracker
```

Original comment by `paul.tot...@gmail.com` on 15 Dec 2006 at 11:23
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed in 1.2.1. Is it?
```

Original comment by `paul.tot...@gmail.com` on 19 Feb 2007 at 9:19

**Comment by GoogleCodeExporter** [comment]
```
All interfaces in version 1.2.2 are defined as in JSR-82
```

Original comment by `skarzhev...@gmail.com` on 16 Mar 2007 at 6:35
- Changed state: **Fixed**
